spec/javascripts/govuk_publishing_components/lib/cookie-settings-spec.js

```diff
@@ -42,157 +43,194 @@ describe('cookieSettings', function () {
+      })
     })
-  })
 
-  describe('showConfirmationMessage', function () {
-    it('sets the previous referrer link if one is present', function () {
-      fakePreviousURL = '/student-finance'
+    describe('showConfirmationMessage', function () {
+      it('sets the previous referrer link if one is present', function () {
+        fakePreviousURL = '/student-finance'
 
-      new GOVUK.Modules.CookieSettings(element).init()
+        new GOVUK.Modules.CookieSettings(element).init()
 
-      var button = element.querySelector('#submit-button')
-      button.click()
+        var button = element.querySelector('#submit-button')
+        button.click()
 
-      var previousLink = document.querySelector('.cookie-settings__prev-page')
+        var previousLink = document.querySelector('.cookie-settings__prev-page')
 
-      expect(previousLink.style.display).toEqual('inline')
-      expect(previousLink.href).toContain('/student-finance')
-    })
+        expect(previousLink.style.display).toEqual('inline')
+        expect(previousLink.href).toContain('/student-finance')
+      })
 
-    it('does not set a referrer if one is not present', function () {
-      fakePreviousURL = null
+      it('does not set a referrer if one is not present', function () {
+        fakePreviousURL = null
 
-      new GOVUK.Modules.CookieSettings(element).init()
+        new GOVUK.Modules.CookieSettings(element).init()
 
-      var button = element.querySelector('#submit-button')
-      button.click()
+        var button = element.querySelector('#submit-button')
+        button.click()
+
+        var previousLink = document.querySelector('.cookie-settings__prev-page')
+
+        expect(previousLink.style.display).toEqual('none')
+      })
+
+      it('does not set a referrer if URL is the same as current page (cookies page)', function () {
+        fakePreviousURL = document.location.pathname
+
+        new GOVUK.Modules.CookieSettings(element).init()
+
+        var button = element.querySelector('#submit-button')
+        button.click()
+
+        var previousLink = document.querySelector('.cookie-settings__prev-page')
+
+        expect(previousLink.style.display).toEqual('none')
+      })
 
-      var previousLink = document.querySelector('.cookie-settings__prev-page')
+      it('shows a confirmation message', function () {
+        var confirmationMessage = document.querySelector('[data-cookie-confirmation]')
 
-      expect(previousLink.style.display).toEqual('none')
+        new GOVUK.Modules.CookieSettings(element).init()
+
+        var button = element.querySelector('#submit-button')
+        button.click()
+
+        expect(confirmationMessage.style.display).toEqual('block')
+      })
     })
+  })
 
-    it('does not set a referrer if URL is the same as current page (cookies page)', function () {
-      fakePreviousURL = document.location.pathname
+  describe('when the single consent api is enabled', function () {
+    beforeEach(function () {
```

Comment:
Is it necessary to set and unset this before/after each test when (I believe) the value isn't manipulated by the tests?

Reply:
I was more wondering if you could set it once for all (although there's only a couple) and unset at the end, rather than before/after each, but I guess it could fall foul of things running in parallel. No big deal either way.

Reply:
I think I'll leave it as it is, if that's okay.